### PR TITLE
Skip uninstall and resource cleanup when fail-fast is enabled.

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by reasonerjt — DISMISSED** [review]

**Comment by blackpiglet** [comment]
@shubham-pampattiwar @kaovilai 
Please also help take a look.

**Review by shubham-pampattiwar — DISMISSED** [review]

File: test/util/velero/velero_utils.go [hunk 1]
```diff
@@ -98,19 +98,19 @@ var pluginsMatrix = map[string]map[string][]string{
 		"csi":     {"velero/velero-plugin-for-csi:v0.6.0"},
 	},
 	"v1.13": {
-		"aws":       {"velero/velero-plugin-for-aws:v1.9.0"},
-		"azure":     {"velero/velero-plugin-for-microsoft-azure:v1.9.0"},
+		"aws":       {"velero/velero-plugin-for-aws:v1.9.2"},
```
Comment (reasonerjt) [inline test/util/velero/velero_utils.go:101]: Ideally the changes in this file should belong to another PR...

File: test/e2e/backup/backup.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func BackupRestoreTest(backupRestoreTestConfig BackupRestoreTestConfig) {
 	})
 
 	AfterEach(func() {
-		if !veleroCfg.Debug {
+		if !CurrentSpecReport().Failed() || !veleroCfg.FailFast {
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply test/e2e/backup/backup.go]: Modified.

File: test/util/velero/velero_utils.go (hunk 1 above)
Comment (blackpiglet) [reply test/util/velero/velero_utils.go]: Moved to a new commit.

**Comment by blackpiglet** [comment]
PR is ready. Please take another look.

**Review by reasonerjt — APPROVED** [review]

**Review by ywk253100 — DISMISSED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by reasonerjt — APPROVED** [review]
